### PR TITLE
Integrate al7079

### PR DESCRIPTION
**Pull request by rgp62**
added batching

### CODE REVIEW

**Comment by dtseidl** [comment]
I'm refactoring the original Al7079 code. Moving some things around to reflect the new structure.

File: cmad/qois/calibration.py [hunk 1]
```diff
@@ -29,3 +31,15 @@ def _qoi(xi, xi_prev, params, u, u_prev,
...
         return 0.5 * jnp.sum(mismatch * mismatch)
+
+class UniaxialCalibration(Calibration):
+    # _qoi not an abstract class, as signatures may be class-specific
+    @staticmethod
+    def _qoi(xi, xi_prev, params, u, u_prev,
+             data_at_step, cauchy_fun, weight) -> Array:
+
+        sig_11 = cauchy_fun(xi, xi_prev, params, u, u_prev)[1,1]
+        pred = jnp.stack([sig_11,xi[2][0]-1.,xi[2][1]-1.],axis=0)
```
Comment (dtseidl) [inline cmad/qois/calibration.py:42]: Indices for the on-axis stress and off-axis strains are hard-coded.

File: cmad/objectives/objective.py [hunk 2]
```diff
@@ -5,13 +5,13 @@
 
 class Objective():
 
-    def __init__(self, qoi, sensitivity_type):
+    def __init__(self, qoi, sensitivity_type, weights=None):
```
Comment (dtseidl) [inline cmad/objectives/objective.py:8]: I want to treat calibration weights more like the data class in terms of their shape/values and evaluation per step in the QoI class. That way they can be specified up front with the data.

File: cmad/calibrations/al7079/hill_calibration_rot.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+qoi = UniaxialCalibration(model, F, data[0], weights_qoi)
+objective = Objective(qoi, sensitivity_type="adjoint gradient", weights=weights_time)
+
+def multiobjective(Rmats, data):
+    def mobjective(varlist):
+        J_total = 0.
+        grad_total = np.zeros(varlist.shape)
+        for j,Rmat in enumerate(Rmats):
+            objective._qoi.model().parameters.set_rotation_matrix(Rmat)
+            objective._qoi._data = data[j]
```
Comment (dtseidl) [inline cmad/calibrations/al7079/hill_calibration_rot.py:101]: Setting a private variable directly from outside the class.
